### DISCUSSION TITLE
State Updates without Emiting

### DISCUSSION THREAD

**Discussion by LeonWehrhahn** [reduxjs/redux#4211] [Help] [answered]
I sometimes find myself wanting to update the store without emiting a new state. For example, I have a large array that gets split up into different arrays. Each sub-array is rendered into component lists. 
For each list item I track if the user has clicked on it. If the user clicks on it, a post is made to the sever through dispatching an event (AsyncThunk). With useAppState the dispatch is awaited and the component changes it's state accordingly. However because I am also updating the store, the whole list is rerendered and my local component state is lost.
Is there a way I can update the store without causing rerenders? Filtering with equalityFn is also not really an option, as I would have to deeply compare a tone of values.

**Comment by markerikson** [comment, marked as answer]
This really sounds like a question of how to design your UI logic and behavior, rather than "update the state without emitting".

Re-rendering the UI based on new Redux state should _not_ wipe out local component state, because that only happens when components are unmounted.  If you're seeing that, it suggests that there are other problems with the way you've set up your UI components.

Can you give examples of how these components are being rendered right now?

**Reply by LeonWehrhahn** [reply]
Okay, thanks for your answer. I went through my UI logic and was able to change it, so that it does not wipe out local component state.
